testsuite/MDAnalysisTests/coordinates/test_trc.py

```diff
@@ -146,44 +141,138 @@ def test_open_twice(self, TRC_U):
+        assert TRC_U.trajectory.periodic is True
+
+    def test_trc_dimensions(self, TRC_U):
+        ts = TRC_U.trajectory[0]
+        assert_allclose(
+            ts.dimensions,
+            [
+                33.72394463,
+                38.87568624,
+                26.74177871,
+                91.316862341,
+                93.911031544,
+                54.514000084,
+            ],
+        )
+
+    def test_trc_distances(self, TRC_U):
+        import MDAnalysis.analysis.distances as distances
+        import MDAnalysis.analysis.atomicdistances as atomicdistances
+
+        ts = TRC_U.trajectory[0]
+
+        atom_1a = TRC_U.select_atoms("resname LYSH and name O and resid 4")
+        atom_1b = TRC_U.select_atoms("resname ARG and name H and resid 3")
+        atom_1c = TRC_U.select_atoms("resname SOLV and name OW and resid 718")
+        atom_1d = TRC_U.select_atoms("resname SOLV and name OW and resid 305")
+
+        atom_2a = TRC_U.select_atoms("resname VAL and name CG1 and resid 1")
+        atom_2b = TRC_U.select_atoms("resname SOLV and name OW and resid 497")
+        atom_2c = TRC_U.select_atoms("resname SOLV and name OW and resid 593")
+        atom_2d = TRC_U.select_atoms("resname SOLV and name OW and resid 497")
+
+        ag1 = atom_1a + atom_1b + atom_1c + atom_1d
+        ag2 = atom_2a + atom_2b + atom_2c + atom_2d
+
+        dist_A = distances.dist(ag1, ag2, box=ts.dimensions)[2]  # return distance
+        dist_B = atomicdistances.AtomicDistances(ag1, ag2, pbc=True).run().results[0]
+
+        assert_allclose(
+            dist_A, [5.9488481, 4.4777278, 20.8165518, 7.5727112], rtol=1e-06
+        )  # Reference values calculated with gromos++ tser
+        assert_allclose(
+            dist_B, [5.9488481, 4.4777278, 20.8165518, 7.5727112], rtol=1e-06
+        )  # Reference values calculated with gromos++ tser
+
+
+class TestTRCReaderTruncOctBox:
+    @pytest.fixture(scope="class")
+    def TRC_U(self):
+        with pytest.raises(NotImplementedError, match="truncated-octahedral"):
+            return mda.Universe(TRC_PDB_VAC, TRC_TRUNCOCT_VAC)
+
+    def test_load_trajectory(self, TRC_U):
+        temp_universe = TRC_U
+        del temp_universe
+
+
+class TestTRCGenboxOrigin:
+    @pytest.fixture(scope="class")
+    def TRC_U(self):
+        with pytest.raises(
+            NotImplementedError, match="doesnt't support a shifted origin!"
+        ):
+            return mda.Universe(TRC_PDB_VAC, TRC_GENBOX_ORIGIN)
+
+    def test_load_trajectory(self, TRC_U):
+        temp_universe = TRC_U
+        del temp_universe
+
+
+class TestTRCGenboxEuler:
+    @pytest.fixture(scope="class")
+    def TRC_U(self):
+        with pytest.raises(
+            NotImplementedError,
+            match=("doesnt't support yawed, pitched or rolled boxes!"),
+        ):
+            return mda.Universe(TRC_PDB_VAC, TRC_GENBOX_EULER)
+
+    def test_load_trajectory(self, TRC_U):
+        temp_universe = TRC_U
+        del temp_universe
```

Comment:
should not be necessary... or did you have a specific reason for including??

Reply:
This was an ugly workaround because the expected error couldn't trigger within a pytest fixture.
I reworked it to be better and to trigger the exception during the generation of the universe.